src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -3098,6 +3326,802 @@ public static Vector128<float> Invoke(Vector128<float> x, Vector128<float> y)
 #endif
         }
 
+        private interface IIndexOfOperator
+        {
+            static abstract int Invoke(ReadOnlySpan<float> result);
+            static abstract int Invoke(Vector128<float> result, Vector128<int> resultIndex);
+            static abstract void Invoke(ref Vector128<float> result, Vector128<float> current, ref Vector128<int> resultIndex, Vector128<int> curIndex);
+            static abstract void Invoke(ref Vector128<float> result, Vector128<float> current, ref Vector128<int> resultIndex, Vector128<int> curIndex, Vector128<float> remainderMask);
+            static abstract int Invoke(Vector256<float> result, Vector256<int> resultIndex);
+            static abstract void Invoke(ref Vector256<float> result, Vector256<float> current, ref Vector256<int> resultIndex, Vector256<int> curIndex);
+            static abstract void Invoke(ref Vector256<float> result, Vector256<float> current, ref Vector256<int> resultIndex, Vector256<int> curIndex, Vector256<float> remainderMask);
+#if NET8_0_OR_GREATER
+            static abstract int Invoke(Vector512<float> result, Vector512<int> resultIndex);
+            static abstract void Invoke(ref Vector512<float> result, Vector512<float> current, ref Vector512<int> resultIndex, Vector512<int> curIndex);
+            static abstract void Invoke(ref Vector512<float> result, Vector512<float> current, ref Vector512<int> resultIndex, Vector512<int> curIndex, Vector512<float> remainderMask);
+#endif
+
+            static virtual float IdentityValue => throw new NotSupportedException();
+        }
+
+        /// <summary>MathF.Max(x, y) (but NaNs may not be propagated)</summary>
+        private readonly struct IndexOfMaxOperator : IIndexOfOperator
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static int Invoke(ReadOnlySpan<float> result)
+            {
+                float curMax = result[0];
+                int curIn = 0;
+                if (float.IsNaN(curMax))
+                {
+                    return curIn;
+                }
+
+                for (int i = 1; i < result.Length; i++)
```

Comment:
I was expecting this to look like https://github.com/dotnet/runtime/pull/93469/files#diff-c9855f90de28628f094b952b640408bbbd586d0d9ac0faf2542b5ba1e289b01fL324 starting at 0 but it seems you changed it a bit.  Was there a reason for this?  Also I see your other implementations start at 0 / MinValue.
I have the same comment for all cases where you changed to start at 1.

Reply:
I should change them all to start at 1 instead of zero, saves 1 itteration of the loop. But based on some comments earlier I will be changing how it does the final step and so will probably be removing the loop entirely. If I don't end up removing the loop I'll make them all start consistently.